controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -463,14 +492,9 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 			Bootstrap: clusterv1.Bootstrap{
 				ConfigRef: bootstrapRef,
 			},
+			FailureDomain: fd,
 		},
 	}
-	machine.Spec.FailureDomain = fd
```

Comment:
Moved setting of the failure domain up into the machine definition instead of setting it afterwards.